tools/qlog/src/streamer.rs

```diff
@@ -54,10 +54,16 @@ pub enum StreamerState {
     Finished,
 }
 
+pub enum SerializationFormat {
```

Comment:
This is a good question. This PR is currently a kitchen sink approach. There's a couple of points to pick at here.

1. We should IMO support both `Trace` and `TraceSeq` in qlog/src/lib.rs. Both of these formats will exist in the qlog specs for the foreseeable. As you correctly point out, this aids deserializers for both formats (which I tested in separate code). 

2a. Keeping `Trace` for serialization purposes is basically a no-op wrt serde. Applications can use the object directly and handle their own serialization and `write()`s themselves, no obligation to use streamer.rs. If they  can live with buffering, things are easy. If they can't or don't want to buffer, the code for streaming writes isn't too hard to come up with but..

2b. streamer.rs is a weird thing anyway. It adds a streaming serializing mechanism for an inherently non-streaming logging format. This helps avoid costly buffering but seems to be working against the design intentions. 

This PR represents the "kitchen sink" approach and doesn't seem to add too much complexity over the hoops the streaming serializer already had to jump through. However, if we accept that streamer.rs made an initial misstep, then there is a good opportunity to simplify the code. Depending on how much we are willing the modify the current behaviour, the following options are available

* streamer.s supports only JSON-SEQ format - avoids internal branching 
* events are only written atomically - simplifies internal state management and makes the streamer.rs usage a bit more friendly
  * in other words, drop the ability to start event, write frames, end event. Instead all events need to be written with their contents. This would require applications to buffer a bit more than they do today. The worst case is a single QUIC packets with lots of frames. That doesn't seem too onerous to me.
  * the upside of making events more atomic is that we avoid problems today when event writes quietly fail because the serializer is stuck in "frame-writing mode". That is an annoying problem we encountered in the quiche library.

My preference is to do both of these things even if it breaks the current API slightly.

Reply:
PR updated to only serialize to JSON-SEQ. The atomic event thing is punted to follow up work.